cdr/resolver.go

```diff
@@ -91,46 +91,43 @@ func parseAnswer(as []answer) ([]*dns.CAA, error) {
 	return rrs, nil
 }
 
-func createClient(timeout, keepAlive time.Duration, proxy string) (*http.Client, string, error) {
+func createClient(proxy string) (*http.Client, string, error) {
 	u, err := url.Parse(proxy)
 	if err != nil {
 		return nil, "", err
 	}
 	transport := &http.Transport{
-		Proxy: http.ProxyURL(u),
-		Dial: (&net.Dialer{
-			Timeout:   timeout,
-			KeepAlive: keepAlive,
-		}).Dial,
-		TLSHandshakeTimeout: 10 * time.Second,
+		Proxy:               http.ProxyURL(u),
+		TLSHandshakeTimeout: 10 * time.Second, // default, doesn't override context
 	}
 	return &http.Client{
 		Transport: transport,
-		Timeout:   timeout,
 	}, u.Host, nil
 }
 
-// CAAPublicResolver holds state needed to talk to GPDNS
-type CAAPublicResolver struct {
+// CAADistributedResolver holds state needed to talk to GPDNS
+type CAADistributedResolver struct {
 	clients     map[string]*http.Client
 	stats       metrics.Scope
 	maxFailures int
+	timeout     time.Duration
+	logger      blog.Logger
 }
 
-// NewCAAPublicResolver returns a initialized CAAPublicResolver
-func NewCAAPublicResolver(scope metrics.Scope, timeout, keepAlive time.Duration, maxFailures int, proxies []string) (*CAAPublicResolver, error) {
-	cpr := &CAAPublicResolver{stats: scope, maxFailures: maxFailures}
+// New returns a initialized CAADistributedResolver
```

Comment:
Also, document what the parameters do. In particular `maxFailures` has a different meaning than in, say, the DNS package.